### PR TITLE
Capture Remote StepAction Location in TaskRun Status

### PR DESCRIPTION
**Pull request by chitrangpatel**
Prior to this, we captured remote Task and pipeline origin information (uri and digest) in the TaskRun/PipelineRun Status. This allowed Chains to pull this information and insert it into the intoto provenance. A similar capability was missing for the newly implemented StepActions. This PR fixes that gap.

Fixes https://github.com/tektoncd/pipeline/issues/8091

# Changes

# Submitter Checklist

As the author of this PR, please check off the items in this checklist:

- [ ] Has [Docs](https://github.com/tektoncd/community/blob/main/standards.md#docs) if any changes are user facing, including updates to minimum requirements e.g. Kubernetes version bumps
- [x] Has [Tests](https://github.com/tektoncd/community/blob/main/standards.md#tests) included if any functionality added or changed
- [x] [pre-commit](https://github.com/tektoncd/pipeline/blob/main/DEVELOPMENT.md#install-tools) Passed
- [x] Follows the [commit message standard](https://github.com/tektoncd/community/blob/main/standards.md#commits)
- [x] Meets the [Tekton contributor standards](https://github.com/tektoncd/community/blob/main/standards.md) (including functionality, content, code)
- [x] Has a kind label. You can add one by adding a comment on this PR that contains `/kind <type>`. Valid types are bug, cleanup, design, documentation, feature, flake, misc, question, tep
- [x] Release notes block below has been updated with any user facing changes (API changes, bug fixes, changes requiring upgrade notices or deprecation warnings). See some examples of [good release notes](https://github.com/tektoncd/community/blob/main/standards.md#good-release-notes).
- [ ] Release notes contains the string "action required" if the change requires additional action from users switching to the new release

# Release Notes

```release-note
Capture Remote StepAction Location in TaskRun Status
```
/kind feature

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -1,6 +1,7 @@
 module github.com/tektoncd/pipeline
 
-go 1.22
+go 1.22.3
```
Comment (dibyom) [inline go.mod:3]: should this match the toolchain version?

Comment (chitrangpatel) [reply go.mod:3]: Hmm. I'm not sure 🤔. This came out of running hack/update-codegen.sh

Comment (chitrangpatel) [reply go.mod]: Fixed. Reverted this change.

**Comment by chitrangpatel** [comment]
@dibyom can you take another look? Thanks!

File: pkg/pod/status.go [hunk 2]
```diff
@@ -139,7 +139,6 @@ func MakeTaskRunStatus(ctx context.Context, logger *zap.SugaredLogger, tr v1.Tas
 	}
 
 	trs.PodName = pod.Name
-	trs.Steps = []v1.StepState{}
```
Comment (dibyom) [inline pkg/pod/status.go:142]: is it necessary to remove this line?

**Review by dibyom — APPROVED** [review]

**Review by JeromeJu — APPROVED** [review]
/lgtm

**Pull request merged**